internal/agent/agent_test.go

```diff
@@ -922,42 +922,64 @@ func Test_ProcessSamplingExcludes(t *testing.T) {
 		ProcessDisplayName: "some-process",
 	}
 
+	boolAsPointer := func(val bool) *bool {
+		return &val
+	}
+
 	type testCase struct {
-		name string
-		c    *config.Config
-		ff   feature_flags.Retriever
-		want bool
+		name          string
+		c             *config.Config
+		ff            feature_flags.Retriever
+		expectInclude bool
 	}
 	testCases := []testCase{
 		{
-			name: "Include not matching must not exclude",
-			c:    &config.Config{IncludeMetricsMatchers: map[string][]string{"process.name": {"does-not-match"}}, DisableCloudMetadata: true},
-			ff:   test.NewFFRetrieverReturning(false, false),
-			want: false,
+			name:          "Include not matching must not include",
```

Comment:
nit: should we add in the name ... with process metrics enabled

Reply:
we have "recreated" all the scenarios in https://github.com/newrelic/infrastructure-agent/pull/1927/files#diff-f54cd787a23de02c24305d62345d1ca23c552309375f9119f7a1adf0badd1c79R1017

we will create a task to clean the old ones